### PR TITLE
use checkPositive/checkPositiveOrZero

### PR DESCRIPTION
**Pull request by thinkerou**
Motivation:

use checkPositive/checkPositiveOrZero

### CODE REVIEW

**Comment by normanmaurer** [comment]
@netty-bot Test this please

File: buffer/src/main/java/io/netty/buffer/PoolArena.java [hunk 1]
```diff
@@ -375,7 +375,8 @@ int alignCapacity(int reqCapacity) {
     }
 
     void reallocate(PooledByteBuf<T> buf, int newCapacity, boolean freeOldMemory) {
-        if (newCapacity < 0 || newCapacity > buf.maxCapacity()) {
+        checkPositiveOrZero(newCapacity, "newCapacity");
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PoolArena.java]: @thinkerou please revert this change... I would prefer to keep this as one if check.

File: codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java [hunk 2]
```diff
@@ -632,15 +630,14 @@ private static void verifyWeight(short weight) {
     }
 
     private static void verifyErrorCode(long errorCode) {
-        if (errorCode < 0 || errorCode > MAX_UNSIGNED_INT) {
+        checkPositiveOrZero(errorCode, "errorCode");
```
Comment (thinkerou) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java]: revert

File: buffer/src/main/java/io/netty/buffer/PoolArena.java (hunk 1 above)
Comment (thinkerou) [reply buffer/src/main/java/io/netty/buffer/PoolArena.java]: done, thanks!

**Comment by thinkerou** [comment]
@normanmaurer done, thanks!

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java [hunk 3]
```diff
@@ -25,6 +27,7 @@
 import io.netty.handler.codec.TooLongFrameException;
 import io.netty.util.ByteProcessor;
 import io.netty.util.internal.AppendableCharSequence;
+import sun.net.www.HeaderParser;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: @thinkerou remove this ?

Comment (thinkerou) [reply codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: oh...sorry need to remove

Comment (thinkerou) [reply codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: @normanmaurer done!

**Comment by normanmaurer** [comment]
Also cherry-picked into 4.1 (4c64c98f348131e0792ba4a92ce3d0003237d56a)
